### DISCUSSION TITLE
Component testing on remote dev container

### DISCUSSION THREAD

**Discussion by wallslide** [cypress-io/cypress#17980] [Component Testing]
I'm trying to develop using github codespaces and cypress component testing. I think the best solution would be to run the test-runner on my local machine and connect it to a remote webpack dev server running on the codespaces devcontainer. Is this possible?

**Comment by herecydev** [comment]
Cypress can talk to any url you've provided, so you would need to expose the port on the codespaces (which is definitely possible).

**Reply by NicholasBoll** [reply]
I think the problem here is the Cypress GUI wants to run against files on a local file system. Cypress can hit any remote URL, but test files are local. What Cypress needs in this case is a Cypress server running on the remote machine and a local GUI hitting that server. So instead of the Cypress GUI requiring local files, connect to a Cypress server to run remote spec files.
